src/UserCollectableAttribute.js

```diff
@@ -101,37 +104,19 @@ class UserColectableAttribute {
     }
   }
 
-  /**
-   * validate the value type
-   * @param {*} value
-   * @param {*} type
-   */
-  static isValueOfType(value, type) {
-    switch (type) {
-      case 'String':
-        return _.isString(value);
-      case 'Number':
-        return _.isNumber(value);
-      case 'Boolean':
-        return _.isBoolean(value);
-      default:
-        return false;
-    }
-  }
-
   static getAllProperties(identifier, pathName) {
     const definition = _.find(definitions, { identifier });
     const properties = [];
-    const type = UserColectableAttribute.resolveType(definition);
+    const type = resolveType(definition, definitions);
```

Comment:
Why do you need to pass the definitions for an UCA file? Wasn't it defined globally?

Reply:
Ignore, now I see that there is custom definitions